options.go

```diff
@@ -635,3 +635,11 @@ func IPv6BlackHoleSuccessCounter(f *swarm.BlackHoleSuccessCounter) Option {
 		return nil
 	}
 }
+
+// ShareTCPListener shares the same listen address between TCP and Websocket transports.
+func ShareTCPListener() Option {
+	return func(cfg *Config) error {
+		cfg.ShareTCPListener = true
+		return nil
+	}
+}
```

Comment:
This API has the nice property of keeping all current code same and enable sharing by just specifying 
```
libp2p.New(
...
ShareTCPListener()
)
```

Reply:
As `libp2p.New` user I will wonder why I need to explicitly pass this Option  to enable sharing for TCP ports, but don't have to do anything extra for UDP ones.

No chance of making this implicit? Or at least explain in godoc comment why this needs to be explicit?

Reply:
The main reason is that I want to get some confidence here before defaulting this behavior to on. In a future release it'll be the default and you wouldn't need this.

Aside: UDP transports don't currently share the port if you use port `0` across them (this does). That's also something that I'd like to change in a future release.

I'll update the comment.